# origin: okir (BitKeeper)
# cset: 1.1521.1.39 (2.4) key=41e83b4apeFNk8VKyuGgYA-vmB8JEg
# URL: http://linux.bkbits.net:8080/linux-2.4/cset@41e83b4apeFNk8VKyuGgYA-vmB8JEg
# inclusion: upstream
# descrition: [NET]: Fix CMSG32_OK macros.
# revision date: Thu, 24 Mar 2005 11:05:35 +0900
#
# S rset: ChangeSet|1.1521.1.38..1.1521.1.39
# I rset: arch/ppc64/kernel/sys_ppc32.c|1.10..1.10.1.1
# I rset: include/asm-x86_64/socket32.h|1.2..1.2.1.1
# I rset: arch/parisc/kernel/sys_parisc32.c|1.3..1.3.1.1
# I rset: arch/ia64/ia32/sys_ia32.c|1.45..1.45.1.1
# I rset: arch/mips64/kernel/linux32.c|1.16..1.16.1.1
# I rset: arch/sparc64/kernel/sys_sparc32.c|1.35..1.35.1.1
# I rset: arch/s390x/kernel/linux32.c|1.15..1.15.1.1
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/14 13:36:10-08:00 okir@suse.de 
#   [NET]: Fix CMSG32_OK macros.
#   
#   Needs to check cmsghdr32 not cmsghdr.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# include/asm-x86_64/socket32.h
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/sparc64/kernel/sys_sparc32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/s390x/kernel/linux32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/ppc64/kernel/sys_ppc32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/parisc/kernel/sys_parisc32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/mips64/kernel/linux32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
# arch/ia64/ia32/sys_ia32.c
#   2005/01/14 13:35:40-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG32_OK macros.
# 
#
===== arch/ppc64/kernel/sys_ppc32.c 1.10 vs 1.10.1.1 =====
--- 1.10/arch/ppc64/kernel/sys_ppc32.c	2004-12-09 06:32:46 +09:00
+++ 1.10.1.1/arch/ppc64/kernel/sys_ppc32.c	2005-01-15 06:35:40 +09:00
@@ -3274,7 +3274,7 @@ asmlinkage long sys32_setsockopt(int fd,
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== include/asm-x86_64/socket32.h 1.2 vs 1.2.1.1 =====
--- 1.2/include/asm-x86_64/socket32.h	2004-12-09 06:32:46 +09:00
+++ 1.2.1.1/include/asm-x86_64/socket32.h	2005-01-15 06:35:40 +09:00
@@ -46,7 +46,7 @@ struct cmsghdr32 {
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== arch/parisc/kernel/sys_parisc32.c 1.3 vs 1.3.1.1 =====
--- 1.3/arch/parisc/kernel/sys_parisc32.c	2004-12-09 06:32:46 +09:00
+++ 1.3.1.1/arch/parisc/kernel/sys_parisc32.c	2005-01-15 06:35:40 +09:00
@@ -1815,7 +1815,7 @@ struct cmsghdr32 {
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== arch/ia64/ia32/sys_ia32.c 1.45 vs 1.45.1.1 =====
--- 1.45/arch/ia64/ia32/sys_ia32.c	2004-12-09 06:32:46 +09:00
+++ 1.45.1.1/arch/ia64/ia32/sys_ia32.c	2005-01-15 06:35:40 +09:00
@@ -1370,7 +1370,7 @@ struct cmsghdr32 {
 	((len) >= sizeof(struct cmsghdr32) ? (struct cmsghdr32 *)(ctl) : (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== arch/mips64/kernel/linux32.c 1.16 vs 1.16.1.1 =====
--- 1.16/arch/mips64/kernel/linux32.c	2004-12-09 06:32:46 +09:00
+++ 1.16.1.1/arch/mips64/kernel/linux32.c	2005-01-15 06:35:40 +09:00
@@ -2484,7 +2484,7 @@ struct cmsghdr32 {
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== arch/sparc64/kernel/sys_sparc32.c 1.35 vs 1.35.1.1 =====
--- 1.35/arch/sparc64/kernel/sys_sparc32.c	2004-12-09 06:32:46 +09:00
+++ 1.35.1.1/arch/sparc64/kernel/sys_sparc32.c	2005-01-15 06:35:40 +09:00
@@ -2355,7 +2355,7 @@ struct cmsghdr32 {
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
===== arch/s390x/kernel/linux32.c 1.15 vs 1.15.1.1 =====
--- 1.15/arch/s390x/kernel/linux32.c	2004-12-09 06:32:46 +09:00
+++ 1.15.1.1/arch/s390x/kernel/linux32.c	2005-01-15 06:35:40 +09:00
@@ -2307,7 +2307,7 @@ struct cmsghdr32 {
 				    (struct cmsghdr32 *)NULL)
 #define CMSG32_FIRSTHDR(msg)	__CMSG32_FIRSTHDR((msg)->msg_control, (msg)->msg_controllen)
 #define CMSG32_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct cmsghdr32) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
